lib/services/profiles-api-service.js

```diff
@@ -98,6 +98,17 @@ function profileApiServiceFactory(
                 }
             });
         }
+
+        if (query.macs && query.ips) {
+            var macAddresses = _.flattenDeep([query.macs]);
+            var ipAddresses = _.flattenDeep([query.ips]);
+            return Promise.map(ipAddresses, function(ip, index) {
+                if (ip) {
+                    return lookupService.setIpAddress(ip, macAddresses[index]);
```

Comment:
Does this call succeed if `macAddresses[n]` is null (I think it will, just double checking)?